vllm/v1/engine/detokenizer.py

```diff
@@ -91,25 +102,34 @@ def from_new_request(
             prompt_token_ids=request.prompt_token_ids,
             tokenizer=tokenizer,
             stop_buffer_length=stop_buffer_length,
-        )
+            logprobs=[] if do_logprobs else None,
```

Comment:
Instead of making `logprobs: Optional[List[SampleLogprobs]]` and using `None` to handle the case where there are no logprobs, you could instead make `logprobs: List[SampleLogprobs]` and treat an empty list as no logprobs

This will simplify the code in the `detokenizer` since you can then avoid the `None` checking

Reply:
In principle I see the value of this. However, I attempted to emulate the behavior expected by the v0 engine unit tests (since there is no reason for the *interface* spec to change between v0 and v1.)

At the link below, I highlighted two lines from a v0 logprobs unit test. The test implicitly configures the number of logprobs as `None` (by leaving logprobs unspecified in the `SamplingParams`) and the test expects that the request output will have `results_logprobs_none[i].outputs[0].logprobs`

https://github.com/vllm-project/vllm/blob/9a99273b482a3e90431069f37858d60827983e2f/tests/samplers/test_logprobs.py#L170-L172

This is why I used slightly more complex logic to make the request output logprobs be `None` when the user does not request logprobs.